aten/src/THNN/generic/SpatialUpSamplingNearest.c

```diff
@@ -2,198 +2,153 @@
 #define TH_GENERIC_FILE "generic/SpatialUpSamplingNearest.c"
 #else
 
+#include "linear_upsampling.h"
+#include <stdio.h>
 
 static inline void THNN_(SpatialUpSamplingNearest_shapeCheck)
      (THTensor *input, THTensor *gradOutput,
-      int scale_factor) {
-  THArgCheck(input != NULL, 2, "4D input tensor expected but got NULL");
-  THArgCheck(scale_factor > 1, 4,
-	     "scale_factor must be greater than 1, but got: %d", scale_factor);
-  THNN_ARGCHECK(input->_dim() == 3 || input->_dim() == 4, 2, input,
-		"3D or 4D input tensor expected but got: %s");
-  if (input->_dim() == 3) {
-    int nChannels    = THTensor_(size)(input, 0);
-    int inputHeight  = THTensor_(size)(input, 1);
-    int inputWidth   = THTensor_(size)(input, 2);
-    int outputHeight = inputHeight * scale_factor;
-    int outputWidth  = inputWidth  * scale_factor;
-    if (gradOutput != NULL) {
-      THNN_CHECK_DIM_SIZE(gradOutput, 3, 0, nChannels);
-      THNN_CHECK_DIM_SIZE(gradOutput, 3, 1, outputHeight);
-      THNN_CHECK_DIM_SIZE(gradOutput, 3, 2, outputWidth);
-    }
-  } else {
-    int nBatch       = THTensor_(size)(input, 0);
-    int nChannels    = THTensor_(size)(input, 1);
-    int inputHeight  = THTensor_(size)(input, 2);
-    int inputWidth   = THTensor_(size)(input, 3);  
-    int outputHeight = inputHeight * scale_factor;
-    int outputWidth  = inputWidth  * scale_factor;
-    if (gradOutput != NULL) {
-      THNN_CHECK_DIM_SIZE(gradOutput, 4, 0, nBatch);
-      THNN_CHECK_DIM_SIZE(gradOutput, 4, 1, nChannels);
-      THNN_CHECK_DIM_SIZE(gradOutput, 4, 2, outputHeight);
-      THNN_CHECK_DIM_SIZE(gradOutput, 4, 3, outputWidth);
-    }
+      int nBatch, int nChannels,
+      int inputHeight, int inputWidth,
+      int outputHeight, int outputWidth) {
+  THArgCheck(inputHeight > 0 && inputWidth > 0
+       && outputHeight > 0 && outputWidth > 0, 2,
+       "input and output sizes should be greater than 0,"
+       " but got input (H: %d, W: %d) output (H: %d, W: %d)",
+       inputHeight, inputWidth, outputHeight, outputWidth);
+  if (input != NULL) {
+    THNN_ARGCHECK(input->_dim() == 4, 2, input,
```

Comment:
Are we dropping support for 3d input?

Reply:
Yes and no ;) My 2cents : I removed it here bc it was actually never triggered due to a dimension guard in python end. https://github.com/pytorch/pytorch/blob/master/torch/nn/functional.py#L1789 And it was never in our python end documentation. 
If we want to support it, we could easily expand the 3D to 4D on python end and pass it along, that way we could avoid complicating our C code by checking sizes. 
What do you think?

Reply:
I'm fine with removing. :)